### ISSUE TITLE
Bun cannot run archiver module

### ISSUE THREAD

**Issue by seveibar** [oven-sh/bun#10416]
### What version of Bun is running?

1.1.4

### What platform is your computer?

Linux 6.8.7-arch1-1 x86_64 unknown

### What steps can reproduce the bug?

> test.ts
```ts
import * as fs from "fs"
import archiver from "archiver"

const output = fs.createWriteStream("./output.zip")
const archive = archiver("zip", { zlib: { level: 9 } })

output.on("close", () => {
  console.log("Zip file created successfully")
})

archive.on("error", (err) => {
  throw err
})

archive.pipe(output)

// Add a test file to the zip
const testFilePath = "./test.txt"
archive.file(testFilePath, { name: "test.txt" })

archive.finalize()
```

> test.txt
```
hello world
```

### What is the expected behavior?

Create archive "output.zip" with "test.txt" inside

### What do you see instead?

The following error:

```
bun test.ts
28 |
29 |   beforeFirstCall(this, '_read', function() {
30 |     var source = fn.call(this, options);
31 |     var emit = this.emit.bind(this, 'error');
32 |     source.on('error', emit);
33 |     source.pipe(this);
         ^
TypeError: undefined is not an object
      at node:stream:2092:83
      at /home/seve/w/tmp/bt1/node_modules/lazystream/lib/lazystream.js:33:5
      at /home/seve/w/tmp/bt1/node_modules/lazystream/lib/lazystream.js:18:5
      at /home/seve/w/tmp/bt1/node_modules/lazystream/node_modules/readable-stream/lib/_stream_readable.js:443:5
      at resume_ (/home/seve/w/tmp/bt1/node_modules/lazystream/node_modules/readable-stream/lib/_stream_readable.js:790:5)
```

### Additional information

- This is currently blocking [tscircuit](https://github.com/tscircuit/tscircuit) from using `bun` as it's default runtime
- [archiver]() has 9m installs weekly

**Comment by vely-digital** [comment]
True, I wanted to transfer some services to bun, but I had this same problem. I hope someone will check this out. Someone reported this same issue last year #6992 .

**Comment by seveibar** [comment]
This appears to be fixed by or before v1.1.7

![image](https://github.com/oven-sh/bun/assets/1910070/9bc81079-e31a-4b4d-b55d-15f12e83537f)

**Comment by LunaFolf** [comment]
Running the same test, I'm unfortunately not experiencing the same outcome.

```
luna@BEEF:~/projects/10416$ bun ./test.ts
28 |
29 |   beforeFirstCall(this, '_read', function() {
30 |     var source = fn.call(this, options);
31 |     var emit = this.emit.bind(this, 'error');
32 |     source.on('error', emit);
33 |     source.pipe(this);
         ^
TypeError: undefined is not an object
      at node:stream:2087:83
      at /home/luna/projects/10416/node_modules/lazystream/lib/lazystream.js:33:5
      at /home/luna/projects/10416/node_modules/lazystream/lib/lazystream.js:18:5
      at /home/luna/projects/10416/node_modules/lazystream/node_modules/readable-stream/lib/_stream_readable.js:443:5
      at resume_ (/home/luna/projects/10416/node_modules/lazystream/node_modules/readable-stream/lib/_stream_readable.js:790:5)

Bun v1.1.7 (Linux x64)
```

the `output.zip` is created, and does contain a `test.txt` however the text file has a size of 0.

Everything I have for a test: https://gist.github.com/LunaFolf/94f8c0a46093028d43d606c54731f857
